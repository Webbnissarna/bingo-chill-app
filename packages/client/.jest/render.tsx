import React from "react";
import { render as rtlRender } from "@testing-library/react";

export function render(ui: React.ReactElement) {
  return rtlRender(<>{ui}</>);
}
