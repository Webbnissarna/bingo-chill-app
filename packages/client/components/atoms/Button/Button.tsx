import React from "react";

interface ButtonProps {
  variant?: "standard" | "dangerous";
  children: React.ReactNode;
  onClick: () => void;
}

export default function Button({
  variant,
  children,
  onClick,
}: ButtonProps): JSX.Element {
  return (
    <button
      className={`font-text font-base text-snowStorm-2 px-3 py-1 rounded-lg transition flex gap-1 items-center justify-center ${
        variant === "dangerous"
          ? "bg-aurora-0 hover:bg-aurora-1 active:bg-aurora-2"
          : "bg-polarNight-3 hover:bg-frost-3 active:bg-frost-2"
      }`}
      onClick={onClick}
    >
      {children}
    </button>
  );
}
