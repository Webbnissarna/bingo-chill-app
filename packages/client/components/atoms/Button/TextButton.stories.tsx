import type { Story, StoryDefault } from "@ladle/react";
import Button from ".";
import { RiseOutlined } from "@ant-design/icons";

interface StoryProps {
  text: string;
}

function StoryComponent({ text }: StoryProps) {
  return (
    <div className="flex flex-col items-start gap-2">
      <Button variant="standard" onClick={() => alert("Clicked!")}>
        {text}
      </Button>
      <Button onClick={() => undefined}>
        <span className="italic">Supports</span>{" "}
        <span className="text-aurora-0">various</span>{" "}
        <span className="font-bold">Content</span>
        <RiseOutlined />
      </Button>
      <Button variant="dangerous" onClick={() => undefined}>
        Dangerous Action
      </Button>
      <Button
        variant="standard"
        disabled
        onClick={() => alert("Should never alert!")}
      >
        Disabled Button
      </Button>
    </div>
  );
}

export const Default: Story<StoryProps> = StoryComponent;

Default.args = {
  text: "Hello World",
};

Default.argTypes = {
  text: {
    control: { type: "text" },
  },
};

export default {
  title: "Atoms / Button",
} satisfies StoryDefault;
