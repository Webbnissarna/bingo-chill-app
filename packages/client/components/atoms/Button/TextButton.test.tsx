import { render } from "../../../.jest/render";
import Button from ".";

describe("Button", () => {
  it("renders", () => {
    const { asFragment } = render(
      <Button text="Hello World" onClick={() => undefined} />,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
