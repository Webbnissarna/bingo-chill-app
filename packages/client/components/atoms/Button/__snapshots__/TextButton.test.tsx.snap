// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders 1`] = `
<DocumentFragment>
  <button
    class="font-text font-base text-snowStorm-2 px-3 py-1 rounded-lg transition flex gap-1 items-center justify-center bg-polarNight-3 hover:bg-frost-3 active:bg-frost-2"
  />
</DocumentFragment>
`;
